api/client/src/main/java/org/projectnessie/client/auth/oauth2/OAuth2Client.java

```diff
@@ -158,6 +158,9 @@ public void close() {
   }
 
   private void wakeUp(Instant now) {
+    if (closing.get()) {
+      return;
```

Comment:
nit: maybe log a debug message in this case?

Reply:
Let me merge this as is and I'll add the debug message in #8338.